doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,368 @@
+.. _cmake_pkg:
+
+Zephyr CMake Package
+####################
+
+Zephyr can be exported as a CMake config package.
+
+A CMake config package allows a developer to integrate Zephyr into their own
+application by using ``find_package(Zephyr)``.
+
+Exporting Zephyr as a CMake package is identical to creating a reference to the
+current Zephyr installation inside the CMake user package registry.
+
+
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      In Linux, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: macOS
+
+      In macOS, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: Windows
+
+      In Windows, the CMake user package registry is found in:
+
+      ``HKEY_CURRENT_USER\Software\Kitware\CMake\Packages\Zephyr``
+
+
+The Zephyr package allows CMake to automatically find a Zephyr base.
+One or more Zephyr installations must be exported.
```

Comment:
Consider adding a comment about this exported installation becoming stale if the detection mechanisms change over time and the exported tree is an old one.

Reply:
Actually I think it is too premature to add this here, let's ignore this comment for now.

Reply:
Functionality explained directly.